src/useField.js

```diff
@@ -180,8 +180,10 @@ const useField = (
     visited: otherState.visited
   }
   if (formatOnBlur) {
-    value = defaultFormat(value, name)
-  } else if (format) {
+    if (component === 'input') {
```

Comment:
This breaks cases where `component` is a custom adapter: #514

Reply:
@erikras any chance you could have a look at this? it's currently preventing us from updating from 4.1 right now :/